runtime/lua/man.lua

```diff
@@ -576,9 +576,8 @@ function M.man_complete(arg_lead, cmd_line, _)
   end
 
   if #args == 1 then
-    -- returning full completion is laggy. Require some arg_lead to complete
-    -- return complete('', '', '')
-    return {}
+    -- returning full completion is laggy. But it is something that users espect (#31512)
+    return complete('', '', '')
```

Comment:
@lewis6991 do you foresee any problems with this? completion is currently broken.

Reply:
Yes, it was very slow.

Reply:
It is slow but not horribly slow, at least on my machine. 

On vim completion with `Man` can lock the UI in this version it is much faster...

Reply:
Depends on your machine. It must have been too slow when I write this.

Reply:
But it's opt-in via `<tab>`, so it's better that it does something than nothing.

Reply:
Not if you use command line completion from nvim-cmp.

Reply:
> Not if you use command line completion from nvim-cmp.

Then they need to deal with that, rather than expect a regression in Nvim core.

Reply:
The problem reported in https://github.com/neovim/neovim/issues/31512 (which also reproduces for me) is not only for the "everything" case. `:Man f<tab>` does not work. Or with any other char.

Reply:
This PR does not fix that. Why would it?

This [PR](https://github.com/neovim/neovim/pull/31571) will probably fix it.